builder/openstack/step_stop_server.go

```diff
@@ -25,27 +25,44 @@ func (s *StepStopServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Stopping server: %s ...", server.ID))
-	if err := startstop.Stop(client, server.ID).ExtractErr(); err != nil {
-		err = fmt.Errorf("Error stopping server: %s", err)
+	// Refresh current server status
+	currentStateRefreshFunc := ServerStateRefreshFunc(client, serverStale)
+	refreshedServer, _, _, err := currentStateRefreshFunc()
+	if err != nil {
+		err = fmt.Errorf("Error refreshing server status: %s", err)
 		state.Put("error", err)
 		return multistep.ActionHalt
 	}
 
-	ui.Message(fmt.Sprintf("Waiting for server to stop: %s ...", server.ID))
-	stateChange := StateChangeConf{
-		Pending:   []string{"ACTIVE"},
-		Target:    []string{"SHUTOFF", "STOPPED"},
-		Refresh:   ServerStateRefreshFunc(client, server),
-		StepState: state,
-	}
-	if _, err := WaitForState(&stateChange); err != nil {
-		err := fmt.Errorf("Error waiting for server (%s) to stop: %s", server.ID, err)
-		state.Put("error", err)
-		ui.Error(err.Error())
-		return multistep.ActionHalt
-	}
+	server := refreshedServer.(*servers.Server)
+	state.Put("server", server)
+	ui.Say(fmt.Sprintf("Refreshed server: %s status: %s", server.ID, server.Status))
+
+	// The server might have already been shut down by Windows Sysprep
+	if server.Status == "SHUTOFF" || server.Status == "STOPPED" {
```

Comment:
Thanks, I have updated my PR so that it should not cause such problem (or maybe I am just failing to recognize it). Could you please take a look if the updated version could cause it as well? It happens much later and I am failing to trace down the specific spot where it happens.